sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -103,6 +104,26 @@ cdef class HistogramBuilder:
         self.ordered_hessians = hessians.copy()
         self.hessians_are_constant = hessians_are_constant
 
+        # list of histograms that can be re-used for other nodes.
+        self.available_histograms = []
+
+    def get_empty_histograms(HistogramBuilder self):
```

Comment:
I don't find "get_empty" very explicit. It's actually "full" with the count values of the past node when recycled.

Maybe `reuse_or_allocate_histogram` would be more explicit.